src/posthog-core.ts

```diff
@@ -764,6 +764,29 @@ export class PostHog {
             this.persistence.set_initial_referrer_info()
         }
 
+        if (event_name === `$set` || event_name === '$set_once') {
+            if (!this._requirePersonProcessing('$set/$set_once')) {
+                return
+            }
+        }
+
+        let usesSetProp = false
+        eachArray([properties?.$set, properties?.$set_once, options?.$set, options?.$set_once], (prop) => {
+            if (prop && !isEmptyObject(prop)) {
+                usesSetProp = true
+            }
+        })
+        if (usesSetProp) {
+            if (!this._requirePersonProcessing('$set/$set_once', 'property')) {
+                properties = extend({}, properties || {})
+                delete properties.$set
+                delete properties.$set_once
+                options = extend({}, options || {})
+                delete options.$set
+                delete options.$set_once
```

Comment:
Not to mention this can be hacked around with [the config option](https://github.com/PostHog/posthog-js/blob/e49ed4a22f9575ee1c0c46a88759005f27a37a89/src/types.ts#L128) which makes these sort of client side protections a little pointless